### PR TITLE
EOS-14872 and EOS-14907 Invalid response code in exceptions

### PR DESCRIPTION
**Pull request by SrikantVishwakarma**
# Backend

## Problem Statement
<pre>
Story Ref (if any): https://jts.seagate.com/browse/EOS-14872, https://jts.seagate.com/browse/EOS-14907
</pre>
## Unit testing on RPM done
<pre>
Yes
</pre>
## Problem Description
<pre>
EOS-14872: GET API for system stats returns 500 for one missing TO and FROM params
EOS-14907: GET API /api/v1/stats returns 500 for invalid values for METRIC and TOTAL_SAMPLE parameter
</pre>
## Solution
<pre>
Raise InvalidRequest exceptions instead CsmInternalError exceptions because parameters are  not valid.
</pre>
## Unit Test Cases
<pre>
Yes
</pre>
Signed-off-by: SrikantVishwakarma <srikant.vishwakarma@seagate.com>

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

File: csm/common/timeseries.py [hunk 1]
```diff
@@ -201,13 +201,13 @@ async def process_request(self, stats_id, panel, from_t, duration_t,
...
             panel = panel.lower()
             metric_list, unit_list = await self._get_metric_list(panel, metric_list, unit)
-            res = await self._aggregate_metric(panel, from_t, duration_t,
-                                            interval, metric_list, query)
-            return await self._convert_payload(res, stats_id, panel, output_format, unit_list)
         except Exception as e:
             Log.debug("Failed to request stats %s" %e)
-            raise CsmInternalError("id: %s, Error: Failed to process timelion "
-                "request %s" %(stats_id,e))
+            raise InvalidRequest("id: %s, Error: Failed to process timelion "
```
Comment (pawankumarsrivastava) [inline csm/common/timeseries.py:206]: Catch both the exceptions InvalidRequest and Base exception. In case of Invalid request raise 400 otherwise 500.

Comment (pawankumarsrivastava) [reply csm/common/timeseries.py:206]: Error message should be generic. Do not explicitly use the names of 3rd party SW

File: csm/core/services/stats.py [hunk 2]
```diff
@@ -131,7 +131,7 @@ async def get_metrics(self, stats_id, metrics_list, from_t, to_t, interval,
                 else:
                     panels[li[0]]["unit"].append(li[2])
         except:
-            raise CsmInternalError("Invalid metric list for Stats %s" %metrics_list)
+            raise InvalidRequest("Invalid metric list for Stats %s" %metrics_list)
```
Comment (pawankumarsrivastava) [inline csm/core/services/stats.py:134]: Catch Invalid request

**Review by pawankumarsrivastava — CHANGES_REQUESTED** [review]

File: csm/core/services/stats.py (hunk 2 above)
Comment (avrnv) [reply csm/core/services/stats.py:134]: @pawankumarsrivastava, InvalidRequest is the CSM exception that is mapped to 400 error code. It's purpose in services is to be raised, not caught. Here and in the snippet above that you've commented the code handles inputs from the request and converts any error occured in process to the InvalidRequest. That's my understanding. Thus I'm OK with these edits.
@SrikantVishwakarma, please, also post your assumptions here.

Comment (pawankumarsrivastava) [reply csm/core/services/stats.py:134]: @avrnv May be my comment was not clear. What I am suggesting is to catch specific expection, in this case it maybe OutofBounds. Don't use except alone.

Comment (SrikantVishwakarma) [reply csm/core/services/stats.py:134]: > caught

I am agree with @avrnv, CSM exceptions are mapped to specific  error codes, and it's purpose is to be raised not caught.

File: csm/common/timeseries.py (hunk 1 above)
Comment (SrikantVishwakarma) [reply csm/common/timeseries.py]: Addressed review comments

File: csm/core/services/stats.py (hunk 2 above)
Comment (SrikantVishwakarma) [reply csm/core/services/stats.py:134]: Addressed  review comments

**Review by pawankumarsrivastava — APPROVED** [review]

**Review by udayan-y21 — CHANGES_REQUESTED** [review]

**Comment by SrikantVishwakarma** [comment]
status_code_fix1

**Comment by SrikantVishwakarma** [comment]
status_code_fix2
status_code_fix3
status_code_fix4

File: csm/common/timeseries.py [hunk 3]
```diff
@@ -255,15 +260,21 @@ async def _parse_interval(self, from_t, duration_t, interval, total_sample):
...
         calculate interval from total_sample
         """
-        diff_sec = int(duration_t) - int(from_t)
+        try:
+            diff_sec = int(duration_t) - int(from_t)
+        except (ValueError, TypeError):
+            raise InvalidRequest(f"from and to time should be integer")
         if diff_sec <= 0:
-            raise CsmInternalError("to time should be grater than from time")
+            raise InvalidRequest("to time should be grater than from time")
```
[the remark this answers is not in the document]
Comment (SrikantVishwakarma) [reply csm/common/timeseries.py]: Addressed review comments.

**Review by udayan-y21 — APPROVED** [review]

**Comment by SrikantVishwakarma** [comment]
Mini-provisioning and VM deployment log files:
[cortx_deployment_EOS-14907_and_EOS-14872.log](https://github.com/Seagate/cortx-manager/files/6421708/cortx_deployment_EOS-14907_and_EOS-14872.log)
[mini_pro_ EOS-14907_and_EOS-14872.log](https://github.com/Seagate/cortx-manager/files/6421709/mini_pro_.EOS-14907_and_EOS-14872.log)

**Comment by SrikantVishwakarma** [comment]
mini_pro_EOS-14907_and_EOS-14872

**Comment by kupranay** [comment]
ok to test

**Pull request merged**